src/NLog/Config/ServiceRepositoryExtensions.cs

```diff
@@ -40,19 +40,21 @@ namespace NLog.Config
 
     internal static class ServiceRepositoryExtensions
     {
-        internal static ServiceRepository GetServiceProvider(this LoggingConfiguration loggingConfiguration)
+        [NotNull]
+        internal static IServiceProvider GetServiceResolver([CanBeNull] this LoggingConfiguration loggingConfiguration)
         {
             return loggingConfiguration?.LogFactory?.ServiceRepository ?? LogManager.LogFactory.ServiceRepository;
         }
 
-        internal static T ResolveService<T>(this ServiceRepository serviceProvider, bool ignoreExternalProvider = true) where T : class
+        [CanBeNull]
+        public static T ResolveService<T>(this IServiceProvider serviceProvider) where T : class
```

Comment:
unneeded change?

Reply:
Nothing wrong with adding `[CanBeNull]`, but you should not remove the parameter `bool ignoreExternalProvider = true`